src/main/java/org/javarosa/xpath/expr/XPathEncryptStringFunc.java

```diff
@@ -0,0 +1,84 @@
+package org.javarosa.xpath.expr;
+
+
+import org.javarosa.core.model.condition.EvaluationContext;
+import org.javarosa.core.model.instance.DataInstance;
+import org.javarosa.xpath.XPathException;
+import org.javarosa.xpath.parser.XPathSyntaxException;
+
+import javax.crypto.Cipher;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.GCMParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.security.SecureRandom;
+import java.util.Base64;
+
+public class XPathEncryptStringFunc extends XPathFuncExpr {
+    public static final String NAME = "encrypt-string";
+    private static final int EXPECTED_ARG_COUNT = 3;
+
+    public XPathEncryptStringFunc() {
+        name = NAME;
+        expectedArgCount = EXPECTED_ARG_COUNT;
+    }
+
+    public XPathEncryptStringFunc(XPathExpression[] args) throws XPathSyntaxException {
+        super(NAME, args, EXPECTED_ARG_COUNT, true);
+    }
+
+    @Override
+    public Object evalBody(DataInstance model, EvaluationContext evalContext, Object[] evaluatedArgs) {
+        return encryptString(evaluatedArgs[0], evaluatedArgs[1], evaluatedArgs[2]);
+    }
+
+    /**
+     * Encrypt a message with the given algorithm and key.
+     *
+     * @param message    a message to be encrypted
+     * @param key        the key used for encryption
+     * @param algorithm  the encryption algorithm to use
+     */
+    private static String encryptString(Object o1, Object o2, Object o3) {
+        String message = FunctionUtils.toString(o1);
+        String key = FunctionUtils.toString(o2);
+        String algorithm = FunctionUtils.toString(o3);
+
+        final String ENCRYPT_ALGO = "AES/GCM/NoPadding";
+        final int TAG_LENGTH_BIT = 128;
+        final int IV_LENGTH_BYTE = 12;
```

Comment:
Question: Isn't it normal to use 16 bytes for the IV on GCM or do I have that wrong?

Reply:
I based this on some example code and didn't know what the rationale for 12 bytes is. However, it seems that 12 bytes is the default IV length for GCM ([stackoverflow](https://stackoverflow.com/questions/31132162/what-size-of-initialization-vector-needed-for-aes-256-encryption-in-java#:~:text=The%20size%20of%20the%20IV%20depends%20on%20the%20mode%2C%20but,the%20default%20is%20highly%20recommended.), [stackexchange](https://crypto.stackexchange.com/questions/41601/aes-gcm-recommended-iv-size-why-12-bytes)).